arkouda/dataframe.py

```diff
@@ -2706,3 +2823,182 @@ def invert_permutation(perm):
+    -------
+    DataFrame
+        Right-Joined Arkouda DataFrame
+    """
+
+    left_cols = left.columns.copy()
+    left_cols.remove(on)
+
+    in_left = inner_join_merge(left, right, on, left_suffix, right_suffix)
+    in_left_cols = in_left.columns.copy()
+    in_left_cols.remove(on)
+
+    not_in_left = right[find(setdiff1d(right[on], left[on]), right[on])]
+    for col in not_in_left.columns:
+        if col in left_cols:
+            new_col = col + right_suffix
+            not_in_left[new_col] = not_in_left[col]
+            not_in_left = not_in_left.drop(col, axis=1)
+
+    nan_cols = list(set(in_left) - set(in_left).intersection(set(not_in_left)))
+
+    for col in nan_cols:
+        # Create a nan array for all values not in the left df
+        nan_arr = zeros(len(not_in_left))
+        nan_arr.fill(np.nan)
+        left_col_type = type(in_left[col])
+        if in_left[col].dtype == int:
+            in_left[col] = akcast(in_left[col], akfloat64)
+        else:
+            nan_arr = akcast(nan_arr, in_left[col].dtype)
+
+        try:
+            not_in_left[col] = left_col_type(nan_arr)
+        except TypeError:
+            not_in_left[col] = nan_arr
+
+    right_ak_df = DataFrame.append(in_left, not_in_left)
+
+    return right_ak_df
+
+
+@typechecked
+def merge(
+    left: DataFrame,
+    right: DataFrame,
+    on: str,
+    how: str,
+    left_suffix: str = "_x",
+    right_suffix: str = "_y",
+) -> DataFrame:
+    """
+    Utilizes the ak.join.inner_join_merge and the ak.join.right_join_merge
+    functions to return a merged Arkouda DataFrame object
+    containing rows from both DataFrames as specified by the merge
+    condition (based on the "how" and "on" parameters).
+    Based on pandas merge functionality.
+    https://github.com/pandas-dev/pandas/blob/main/pandas/core/reshape/merge.py#L137
+    Parameters
+    ----------
+    left: DataFrame
+        The Left DataFrame to be joined
+    right: DataFrame
+        The Right DataFrame to be joined
+    on: str
+        The name of the DataFrame column the join is being
+        performed on
+    how: str
+        The merge condition.
+        Must be "inner", "left", or "right"
+    left_suffix: str = "_x"
+        A string indicating the suffix to add to columns from the left dataframe for overlapping
+        column names in both left and right. Defaults to "_x". Only used when how is "inner"
+    right_suffix: str = "_y"
+        A string indicating the suffix to add to columns from the right dataframe for overlapping
+        column names in both left and right. Defaults to "_y". Only used when how is "inner"
+    Returns
+    -------
+    DataFrame
+        Joined Arkouda DataFrame
+    """
+
+    if how == 'inner':
+        return inner_join_merge(left, right, on)
+    elif how == 'right':
+        return right_join_merge(left, right, on)
+    elif how == 'left':
+        return right_join_merge(right, left, on, left_suffix='_y', right_suffix='_x')
```

Comment:
i have to imagine we also we also want to pass in `left_suffix` and `right_suffix` in the `'inner'` and `'right'` join cases and also use what is passed in, instead of hardcoding in the `'left'` join case. What is here works fine if everything stays the default but will ignore any attempt by the user to change these suffixes. I'll go ahead and add these as a suggestion and commit them in the interest of getting this in for this release. If this isn't what u intended, it should be pretty easy to change
```suggestion
    if how == 'inner':
        return inner_join_merge(left, right, on, left_suffix, right_suffix)
    elif how == 'right':
        return right_join_merge(left, right, on, left_suffix, right_suffix)
    elif how == 'left':
        return right_join_merge(right, left, on, right_suffix, left_suffix)
```